src/main/java/frc/utils/joysticks/SmartJoystick.java

```diff
@@ -0,0 +1,212 @@
+     *       *****           *****                    *****            *****
+     *           *************                            *************
+     * @formatter:on
+     * </p>
+     */
+    public double getSquaredAxis(Axis axis) {
+        if (joystick == null) {
+            return 0;
+        }
+        if (!isStickAxis(axis)) {
+            return axis.getValue(this);
+        }
+        double squaredAxisValue = getAxisValue(axis) * SmartJoystickConstants.JOYSTICK_AXIS_TO_SQUARE_FACTOR;
+        return MathUtil.clamp(squaredAxisValue, -1, 1);
+    }
+
+    /**
+     * Make the stick value be parabolic instead of linear. By that it gives easier and soft control in low values.
+     *
+     * @param axis - axis the take value from
+     * @return the soft value
+     */
+    public double getSensitiveJoystickValue(Axis axis) {
+        return getSensitiveJoystickValue(getAxisValue(axis), SmartJoystickConstants.SENSITIVE_AXIS_VALUE_POWER);
+    }
+
+    private double getSensitiveJoystickValue(double axisValue, double power) {
+        return Math.pow(Math.abs(axisValue), power) * Math.signum(axisValue);
+    }
+
+    public double getSquaredSensitiveAxis(Axis axis) {
+        double squaredValue = getSquaredAxis(axis);
+        return getSensitiveJoystickValue(squaredValue, SmartJoystickConstants.SENSITIVE_SQUARED_AXIS_VALUE_POWER);
+    }
+
+    private boolean isStickAxis(Axis axis) {
+        return (axis != Axis.LEFT_TRIGGER) && (axis != Axis.RIGHT_TRIGGER);
+    }
+
+    /**
+     * This function returns a joystick
+     *
+     * @return The joystick used in this class.
+     */
+    public Joystick getRawJoystick() {
+        return joystick;
+    }
+
+    public double getAxisValue(Axis axis) {
+        return isStickAxis(axis) ? deadzone(axis.getValue(this)) : axis.getValue(this);
+    }
+
+    private double deadzone(double power) {
+        return MathUtil.applyDeadband(power, deadzone);
+    }
+
+    /**
+     * Rumbles the controller at a certain side
+     *
+     * @param left rumble side (false for left)
+     * @param power normalized rumble [0, 1]
+     */
+    public void rumble(boolean left, double power) {
+        joystick.setRumble(left ? GenericHID.RumbleType.kLeftRumble : GenericHID.RumbleType.kRightRumble, power);
+    }
+
+    public enum Axis {
+        LEFT_X(0, true),
+        LEFT_Y(1, true),
+        LEFT_TRIGGER(2, false),
+        RIGHT_TRIGGER(3, false),
+        RIGHT_X(4, false),
+        RIGHT_Y(5, true);
+
+        private final int axis;
+
+        private int inverted;
+
+        Axis(int axis, boolean isInverted) {
+            this.axis = axis;
+            setInverted(isInverted);
+        }
+
+        public void setInverted(boolean isInverted) {
+            inverted = isInverted ? -1 : 1;
+        }
+
+        public double getValue(SmartJoystick stick) {
+            return inverted * stick.getRawAxis(axis);
+        }
+    }
```

Comment:
maybe, move it out of the class make protected and insert asButton function in axis instead of in smartJoystick